build.gradle

```diff
@@ -23,6 +23,12 @@ apply plugin: 'java'
 apply plugin: 'com.palantir.baseline'
 apply plugin: 'com.palantir.consistent-versions'
 apply plugin: 'com.palantir.git-version'
+apply plugin: 'com.palantir.baseline-java-versions'
+
+javaVersions {
+    libraryTarget = 11
+    runtime = 17
```

Comment:
Will this use JDK 17 for compilation & test running, use JDK 11 source level language features, and generate JDK 11 compatible byte code (version 55.0), is that correct? Does it use JDK 11 or 17 ABI?

Reply:
We're actually using the full toolchain at the requested language level for compilation, so in this case gradle is downloading a JDK11 which is used for compilation (in practice it uses the detected jdk11 on the CI docker image) and downloading a JDK17 to execute tests with the compiled artifact.

in cjr-api I ended up bumping the compile target because we'd been producing Java 8 bytecode (52) while relying on Java 11 dependencies, this solution detects when incompatible bytecode is present in dependencies. https://github.com/palantir/conjure-java-runtime-api/pull/714

Reply:
Ahh nice, best of both worlds